packages/http-client-java/generator/http-client-generator-mgmt/src/main/java/com/microsoft/typespec/http/client/generator/mgmt/model/clientmodel/fluentmodel/ResourceOperation.java

```diff
@@ -175,12 +177,10 @@ protected List<FluentCollectionMethod> getMethodReferencesOfFullParameters() {
 
     protected Optional<FluentCollectionMethod> findMethod(boolean hasContextParameter,
         List<ClientMethodParameter> parameters) {
-        Optional<FluentCollectionMethod> methodOpt = this.getMethodReferencesOfFullParameters()
-            .stream()
-            .filter(m -> hasContextParameter
-                ? m.getInnerClientMethod().getParameters().stream().anyMatch(FluentUtils::isContextParameter)
-                : m.getInnerClientMethod().getParameters().stream().noneMatch(FluentUtils::isContextParameter))
-            .findFirst();
+        Optional<FluentCollectionMethod> methodOpt = findMethodCandidates(hasContextParameter)
+            .filter(method -> !method.getInnerClientMethod().getType().name().endsWith(Utils.METHOD_TYPE_REST_RESPONSE))
+            .findFirst()
+            .or(() -> findMethodCandidates(hasContextParameter).findFirst());
```

Comment:
This is for fluent method(e.g. `apply`, `create`) implementation.
We could match both `updateWithResponse` and `update`. In which case, we need the LRO variant `update` of the collection methods for implementation.

Reply:
query: Does it mean we want to find the Sync version of the method?

I thought we had `update` and `updateWithResponse` in sync-via-async as well? What's the difference?

Also better to put something to code comment.

Reply:
> query: Does it mean we want to find the Sync version of the method?

Yes. 

> I thought we had update and updateWithResponse in sync-via-async as well? What's the difference?

Guess it's added before I made `XXWithResponse` of LRO method private visibility. Now I changed the logic to filter only public visibility.